### PR TITLE
Document capability_info, refactor texture info slightly

### PR DESCRIPTION
**Pull request by kainino0x**
## Dependent on #574

<hr>

**Author checklist for test code/plans:**

- [ ] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

**Comment by kainino0x** [comment]
@shrekshao PTAL at the changes to `src/webgpu/capability_info.ts` in the last commit. (Rest of the commits are just the same ones from #574). You can ignore the rest of the changes, as they're trivial and checked by the typescript compiler.

**Comment by kainino0x** [comment]
Actually let me split the last commit into two.

**Comment by kainino0x** [comment]
There, caa38cefa4aa6951bd0db67739c8570d8c5b3597 is the trivial change, fd6ebcc2b5e7f76b76bf6f842e9c52e03ba98545 adds the documentation

**Comment by kainino0x** [comment]
And e6a15a01ee02b1c438faa55ebabfa09546bad190 adds a little bit more documentation.

File: docs/helper_index.txt [hunk 1]
```diff
@@ -54,7 +54,6 @@ Whenever a new generally-useful helper is added, it should be indexed here.
     (like in copyBufferToTexture, copyTextureToBuffer, writeTexture).
 - {@link webgpu/util/texture/subresource}: Helpers for working with texture subresource ranges.
 - {@link webgpu/util/texture/texel_data}:
```
Comment (shrekshao) [inline docs/helper_index.txt:56]: Seems missing a description

**Review by shrekshao — APPROVED** [review]
LGTM. tsdoc page looks good as well

**Pull request merged**